effekt/shared/src/main/scala/effekt/Parser.scala

```diff
@@ -337,8 +337,20 @@ class Parser(tokens: Seq[Token], source: Source) {
   def stmt(): Stmt =
     nonterminal:
       {
-        if peek(`{`) then BlockStmt(braces { stmts(inBraces = true) }, span())
-        else when(`return`) { Return(expr(), span()) } { Return(expr(), span()) }
+        if (peek(`{`)) {
+          // Ambiguity: `{ ... }` could be a block statement OR a block literal used as UFCS receiver
+          // Disambiguate: if the `}` is followed by `.`, parse as expression (UFCS)
+          backtrack {
+            val blk = functionArg()
+            if (!peek(`.`)) fail("Expected `.` for block UFCS")
+            val e = callExprContinuation(blk)
+            Return(e, span())
+          } getOrElse {
+            BlockStmt(braces { stmts(inBraces = true) }, span())
+          }
```

Comment:
This sucks because currently most occurrences of `{` are, in fact, a BlockStmt. (so it feels like the default should be a "normal" block stmt, not a function arg...)
Perhaps this could be somehow refactored by sharing more stuff with `functionArg`?